nano/node/json_handler.cpp

```diff
@@ -169,18 +169,12 @@ void nano::json_handler::response_errors ()
 	}
 	if (ec)
 	{
-		boost::property_tree::ptree response_error;
-		response_error.put ("error", ec.message ());
-		std::stringstream ostream;
-		boost::property_tree::write_json (ostream, response_error);
-		response (ostream.str ());
-	}
-	else
-	{
-		std::stringstream ostream;
-		boost::property_tree::write_json (ostream, response_l);
-		response (ostream.str ());
+		response_l.put ("error", ec.message ());
```

Comment:
This is changing the behaviour of `nano::json_handler::response_errors ()` to allow for returning an error mixed with partial results. Previously, error was always returned alone, even if partial results existed. Since response_errors () is called in many places, and it is difficult to check all of them, I suggest we do not change its behaviour and instead create a special case function for returning error with partial results (if that is what we decide to do). For example, `nano::json_handler::response_errors_with_data ()`.

Reply:
I put it in a dedicated commit to make this change in behavior clear.
I think that the error "Account not found" is incorrect in this case and it should be "Account not opened"

We could have an additional flag `partial_data` in `account_info` rpc call.
This way the end-user would make it clear he wants to be shown partial data for this call and it would be easy on integrations. Also it would limit the impact on this rpc call alone.